topics/transactions.md

```diff
@@ -120,10 +120,37 @@ MULTI
 +OK
 INCR a b c
 -ERR wrong number of arguments for 'incr' command
+EXEC
+-EXECABORT Transaction discarded because of previous errors.
 ```
 
 This time due to the syntax error the bad `INCR` command is not queued
-at all.
+at all. And the `EXEC` command will receive an `EXECABORT` error.
+
+There are some more specific scenarios: in the `MULTI` context, commands are
+successfully queued (i.e., `QUEUED` reply is received), but when the `EXEC` command
+is executed, it is found that the data needed for these commands does not belong to
+the current node (for example, in cluster mode, the accessed slot has been migrated to
+another node; in standalone mode, a primary-replica switch has occurred). In this case,
+the `EXEC` command will receive a `MOVED` or `REDIRECT` result.
```

Comment:
```suggestion
When the `EXEC` command is processed, the server will check if a failover or slot migration has occurred since queuing the commands.
If either event has occurred, a `-MOVED` or `-REDIRECT` error will be returned if needed without processing the transaction.
```

I think just being a bit more succinct for what the user will see might make the English problem easier?

Reply:
can we also clarify when `-MOVED`/`-REDIRECR` would be expected as well? I feel like the table (option 1) at https://github.com/valkey-io/valkey/pull/895#issuecomment-2314815331 is very helpful. It would be great if it could be incorporated in this doc.